packages/commonwealth/client/scripts/models/Reaction.ts

```diff
@@ -17,28 +17,27 @@ class Reaction {
 
   constructor({
     id,
-    author,
-    chain,
+    Address,
     reaction,
-    threadId,
-    proposalId,
-    commentId,
+    thread_id,
+    proposal_id,
+    comment_id,
     author_chain,
-    canvasAction,
-    canvasSession,
-    canvasHash,
+    canvas_action,
+    canvas_session,
+    canvas_hash,
```

Comment:
There were some `modelFromServer` methods in each legacy state controller, which were basically doing some calculations to API responses and then passing them to  `client/scripts/models/{_relevant model_}` -- I dealt with many issues in this approach since instead of a unified transformer we had 2 separate ones each doing something a bit different, this was also due to the fact that we were using object `[]` as state and when updating state we were not modeling it from server response (mostly).

Now the updating working of `client/scripts/models/{_relevant model_}` (the ones which I updated at least) is a bit different and more unified -- also due to our state being in react query. The snake_cases you see above are coming from the API response, so I wouldn't really bother about those but curious if you have any suggestions to improve the response transformation here

Reply:
got ya, well, sharing model for front end and back end is always problematic. For sure not the scope of this PR, but would be nice to think about it in the near future. In graphQL it would be a bit easier to share/generate types but not a case for us. Heard good things about https://trpc.io/ recently. We have monorepo so this fits perfectly for us in this case, might be worth checking.

cc @rbennettcw @jnaviask

Reply:
[Followup here](https://github.com/hicommonwealth/commonwealth/pull/4588#issuecomment-1665613964)